src/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -532,6 +532,10 @@ public long Address
                 {
                     if (PrivateAddress != value)
                     {
+                        if (this is ReadOnlyIPAddress)
+                        {
+                            throw new SocketException(SocketError.OperationNotSupported);
```

Comment:
@nil4 , @stephentoub 
Is `SocketException` the best option here? Given the example at https://github.com/dotnet/corefx/issues/33373#issuecomment-438515269 it seems a little out of context.

Reply:
It's what's already thrown when you try to set the address of an IPv6 IPAddress, and it's the only exception type documented for IPAddress.Address to throw.

Reply:
@alfredmyers I kept it consistent with the error thrown above 

https://github.com/dotnet/corefx/blob/ba6a1037c4e3dd0f0cc46a02a8a2b649e6adec7c/src/System.Net.Primitives/src/System/Net/IPAddress.cs#L539-L542

where changing an IPv6 `Address` value is blocked. My assumption was that this exception would already be documented, and did not want to introduce another type (and error message string) for blocking the same thing here.

Reply:
LGTM